### PR TITLE
Remove Deprecated SimpleFS

### PR DESCRIPTION
**Pull request by nknize**
Lucene 9 removes support for SimpleFS File System format. This PR completely
removes SimpleFS support which was deprecated in a previous PR.

### CODE REVIEW

File: server/src/main/java/org/opensearch/index/store/FsDirectoryFactory.java [hunk 1]
```diff
@@ -104,15 +100,8 @@ protected Directory newFSDirectory(Path location, LockFactory lockFactory, Index
                 }
             case MMAPFS:
                 return setPreload(new MMapDirectory(location, lockFactory), lockFactory, preLoadExtensions);
+            // simplefs was removed in Lucene 9; support for enum is maintained for bwc
             case SIMPLEFS:
```
Comment (reta) [inline server/src/main/java/org/opensearch/index/store/FsDirectoryFactory.java:104]: Should we throw the exception here that `SIMPLEFS` is not supported anymore?

Comment (Bukhtawar) [reply server/src/main/java/org/opensearch/index/store/FsDirectoryFactory.java:104]: Maybe add an `assert` instead as an invariant?

Comment (nknize) [reply server/src/main/java/org/opensearch/index/store/FsDirectoryFactory.java:104]: It's not needed here since this is for bwc only.

**Review by reta — APPROVED** [review]

File: server/src/main/java/org/opensearch/index/store/FsDirectoryFactory.java (hunk 1 above)
Comment (nknize) [reply server/src/main/java/org/opensearch/index/store/FsDirectoryFactory.java:104]: The deprecation logging is already handled in [MetadataUpdateSettingsService](https://github.com/opensearch-project/OpenSearch/blob/77d9b643b291b56d503abd84778ad533d0e98076/server/src/main/java/org/opensearch/cluster/metadata/MetadataUpdateSettingsService.java#L288)

**Review by adnapibar — APPROVED** [review]

**Pull request merged**